### PR TITLE
RC-0.1

### PR DESCRIPTION
**Pull request by zavarin-michael**

### CODE REVIEW

File: open_schools_platform/common/tests/models/test_random_model.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from datetime import timedelta
+
+from django.test import TestCase
+from django.utils import timezone
+from django.db.utils import IntegrityError
+
+from open_schools_platform.common.models import RandomModel
+
+
+class RandomModelTests(TestCase):
```
Comment (IlyaBat9) [inline open_schools_platform/common/tests/models/test_random_model.py:10]: Send this to the docs?

File: open_schools_platform/common/tests/services/test_model_update.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import unittest
+from unittest.mock import Mock
+
+from open_schools_platform.common.services import model_update
+
+
+class ModelUpdateTests(unittest.TestCase):
```
Comment (IlyaBat9) [inline open_schools_platform/common/tests/services/test_model_update.py:7]: This is docs code

File: open_schools_platform/organization_management/employees/services.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        raise PermissionDenied()
+
+    pwd = generate_user_password()
+    add_user = get_user(filters={"phone": phone})
+    if not add_user:
+        msg = OrganizationConstants.get_invite_message(phone=phone, pwd=pwd)
+        response = send_sms(to=[phone], msg=msg)
+
+        if response[str(phone)] != 100:
+            raise TimeoutErrorException(detail="Something wrong! Please, contact the administrator"
```
Comment (IlyaBat9) [inline open_schools_platform/organization_management/employees/services.py:39]: Add administrator or support email from .env file?

File: open_schools_platform/user_management/authentication/apis.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+from open_schools_platform.api.swagger_tags import SwaggerTags
+from ..users.serializers import UserSerializer
+
+
+class UserSessionLoginApi(APIView):
+    """
+    Following https://docs.djangoproject.com/en/3.1/topics/auth/default/#how-to-log-a-user-in
+    """
+    class InputSerializer(serializers.Serializer):
+        email = serializers.EmailField()
```
Comment (IlyaBat9) [inline open_schools_platform/user_management/authentication/apis.py:29]: email?

File: open_schools_platform/user_management/authentication/tests/apis/test_user_login.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        self.jwt_logout_url = reverse('api:user-management:authentication:jwt:logout')
+        self.me_url = reverse('api:user-management:authentication:me')
+
+    def test_non_existing_user_cannot_login(self):
+        self.assertEqual(0, User.objects.count())
+
+        data = {
+            "phone": "+79222112943",
+            "password": "qwe",
+            "name": "Vasya",
```
Comment (IlyaBat9) [inline open_schools_platform/user_management/authentication/tests/apis/test_user_login.py:25]: Name

File: open_schools_platform/user_management/authentication/tests/apis/test_user_login.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+        self.jwt_login_url = reverse('api:user-management:authentication:jwt:login')
+        self.jwt_logout_url = reverse('api:user-management:authentication:jwt:logout')
+        self.me_url = reverse('api:user-management:authentication:me')
+
+    def test_non_existing_user_cannot_login(self):
+        self.assertEqual(0, User.objects.count())
+
+        data = {
+            "phone": "+79222112943",
```
Comment (IlyaBat9) [inline open_schools_platform/user_management/authentication/tests/apis/test_user_login.py:23]: Phone number?

File: open_schools_platform/user_management/authentication/tests/apis/test_user_login.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        """
+        1. Create user
+        2. Assert login is OK
+        3. Call /api/auth/me
+        4. Assert valid response
+        """
+        credentials = {
+            "phone": "+79222112943",
+            "password": "qwe",
+            "name": "Vasya",
```
Comment (IlyaBat9) [inline open_schools_platform/user_management/authentication/tests/apis/test_user_login.py:42]: ^^

File: open_schools_platform/user_management/users/tests/services/test_user_create.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+            name="Ivan",
+            password="qwe",
+        )
+
+        with self.assertRaises(ValidationError):
+            create_user(
+                phone="+791121129",
+                name="Ivan",
+                password="qwe",
+            )
```
Comment (IlyaBat9) [inline open_schools_platform/user_management/users/tests/services/test_user_create.py:21]: ^^

**Review by IlyaBat9 — CHANGES_REQUESTED** [review]
Hide files from template to docs or no?

File: config/django/base.py [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = environ.Path(__file__) - 3
+
+env.read_env(os.path.join(BASE_DIR, ".env"))
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = '=ug_ucl@yi6^mrcjyz%(u0%&g2adt#bz3@yos%#@*t#t!ypx=a'
```
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply config/django/base.py:26]: It is OK. In production settings we get secret_key from .env file

File: config/django/base.py [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply config/django/base.py:29]: It is OK. In production settings we run in False mode.

File: config/django/base.py [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+
+ALLOWED_HOSTS = ['*']
```
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply config/django/base.py:31]: It is probably OK. In production settings we get list of host from .env file

File: config/django/base.py [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+        },
+    },
+]
+
+WSGI_APPLICATION = 'config.wsgi.application'
+
+# Database
+# https://docs.djangoproject.com/en/3.0/ref/settings/#databases
+
+DATABASES = {
```
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply config/django/base.py:108]: Why?

File: docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,67 @@
+version: "3.9"
+
+services:
+  db:
+    image: postgres:14.1
+    environment:
+      - POSTGRES_DB=local_db
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
```
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply docker-compose.yml:9]: ^^
Maybe we could use .env variables there

File: open_schools_platform/common/tests/models/test_random_model.py (hunk 1 above)
Comment (zavarin-michael) [reply open_schools_platform/common/tests/models/test_random_model.py:10]: I decided to delete this test. I think this code would be useful for private docs, not public .

File: open_schools_platform/common/tests/services/test_model_update.py (hunk 2 above)
Comment (zavarin-michael) [reply open_schools_platform/common/tests/services/test_model_update.py:7]: No, it isn't

File: config/django/base.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (IlyaBat9) [reply config/django/base.py:26]: Ok.

**Review by IlyaBat9 — CHANGES_REQUESTED** [review]
A little bit more

File: open_schools_platform/user_management/authentication/serializers.py [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+class UserUpdateSerializer(serializers.Serializer):
+    name = serializers.CharField(
+        allow_null=False,
+        allow_blank=False
+    )
+
+
+class PasswordUpdateSerializer(serializers.Serializer):
+    old_password = serializers.CharField(min_length=6, max_length=40)
+    new_password = serializers.CharField(min_length=6, max_length=40)
```
[the remark this answers is not in the document]
Comment (Ivan228777) [reply open_schools_platform/user_management/authentication/serializers.py:35]: This field is not about password confirmation. This is serialiser for api request for updating user password.
So, for updating password user should provide correct current password (old_password from his account) and new one (new password)
So this is ok

**Review by IlyaBat9 — APPROVED** [review]

**Pull request merged**